pkg/storage/client_merge_test.go

```diff
@@ -186,25 +187,25 @@ func TestStoreRangeMergeWithData(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
 	for _, colocate := range []bool{false, true} {
-		for _, retries := range []int{0, 3} {
+		for _, retries := range []int64{0, 3} {
 			t.Run(fmt.Sprintf("colocate=%v/retries=%d", colocate, retries), func(t *testing.T) {
 				mergeWithData(t, colocate, retries)
 			})
 		}
 	}
 }
 
-func mergeWithData(t *testing.T, colocate bool, retries int) {
+func mergeWithData(t *testing.T, colocate bool, retries int64) {
 	ctx := context.Background()
 	sc := storage.TestStoreConfig(nil)
 	sc.TestingKnobs.DisableReplicateQueue = true
 
 	// Maybe inject some retryable errors when the merge transaction commits.
 	sc.TestingKnobs.TestingRequestFilter = func(ba roachpb.BatchRequest) *roachpb.Error {
-		if retries > 0 {
+		if atomic.LoadInt64(&retries) > 0 {
 			for _, req := range ba.Requests {
 				if et := req.GetEndTransaction(); et != nil && et.InternalCommitTrigger.GetMergeTrigger() != nil {
-					retries--
+					atomic.AddInt64(&retries, -1)
```

Comment:
Do we want some kind of CAS approach here to avoid letting `retries` drop below 0? Does that matter?